utils/joiner/tuplejoiner.cpp

```diff
@@ -199,6 +235,166 @@ bool TupleJoiner::operator<(const TupleJoiner& tj) const
     return size() < tj.size();
 }
 
+void TupleJoiner::getBucketCount()
+{
+    // get the # of cores, round up to nearest power of 2
+    // make the bucket mask
+    numCores = sysconf(_SC_NPROCESSORS_ONLN);
+    if (numCores <= 0)
+        numCores = 8;
+    bucketCount = (numCores == 1 ? 1 : (1 << (32 - __builtin_clz(numCores - 1))));
+    bucketMask = bucketCount - 1;
+}
+
+template<typename buckets_t, typename hash_table_t>
+void TupleJoiner::bucketsToTables(buckets_t *buckets, hash_table_t *tables)
+{
+    uint i;
+
+    bool done = false, wasProductive;
```

Comment:
I like the name of the variable - wasProductive :)

Reply:
:dancer: